lib/relooper/src/graph/cfg/mod.rs

```diff
@@ -1,89 +1,136 @@
-use crate::graph::cfg::CfgEdge::{Cond, Terminal, Uncond};
+use crate::graph::cfg::CfgEdge::{Cond, Switch, Terminal, Uncond};
 use crate::traversal::graph::bfs::Bfs;
-use std::collections::{HashMap, HashSet};
+use std::collections::{HashMap, HashSet, VecDeque};
 use std::fmt::Debug;
 use std::hash::Hash;
 
 mod cfg_parsing;
 
-pub trait CfgLabel: Copy + Hash + Eq + Ord + Sized {}
+pub trait CfgLabel: Copy + Hash + Eq + Ord + Debug {}
 
-impl<T: Copy + Hash + Eq + Ord + Sized> CfgLabel for T {}
+impl<T: Copy + Hash + Eq + Ord + Debug> CfgLabel for T {}
 
-#[derive(Copy, Clone, PartialEq, Eq, Hash, Debug)]
+#[derive(Clone, PartialEq, Eq, Hash, Debug)]
 pub enum CfgEdge<TLabel> {
     Uncond(TLabel),
     Cond(TLabel, TLabel),
+    Switch(Vec<(usize, TLabel)>),
     Terminal,
 }
 
 impl<TLabel> CfgEdge<TLabel> {
     pub fn iter(&self) -> CfgEdgeIter<&TLabel> {
         match self {
-            Self::Uncond(u) => CfgEdgeIter {
+            Uncond(u) => CfgEdgeIter::Fixed(CfgEdgeFixedIter {
                 inner: [Some(u), None],
                 index: 0,
-            },
-            Self::Cond(cond, fallthrough) => CfgEdgeIter {
+            }),
+            Cond(cond, fallthrough) => CfgEdgeIter::Fixed(CfgEdgeFixedIter {
                 inner: [Some(cond), Some(fallthrough)],
                 index: 0,
-            },
-            Self::Terminal => CfgEdgeIter {
+            }),
+            Switch(v) => {
+                if v.len() <= 2 {
+                    let inner = [v.get(0).map(|(_, l)| l), v.get(1).map(|(_, l)| l)];
+                    CfgEdgeIter::Fixed(CfgEdgeFixedIter { inner, index: 0 })
+                } else {
+                    CfgEdgeIter::Flexible(v.iter().map(|(_, l)| l).collect())
```

Comment:
I don't really like needing to make another heap-allocated object for this. I feel like there is a better way. I'll play around with it a bit and maybe make a suggestion.

Reply:
the problem here is such switch forms can contain really huge number of branches (number of jumpdest instructions, for example)
also, I dont like code duplication with `iter`/`into_iter` functions

Reply:
https://github.com/MCJOHN974/evm2near/pull/19